### ISSUE TITLE
FEC Design Process Walkthrough

### ISSUE THREAD

**Issue by porta-antiporta** [18F/fec-transition#54]
Once we have the folks from the FEC identified for supporting design, we should have a kickoff meeting to go over what design activities are present for the project and identify where folks can participate.

- [X] Identify folks (Jonella C., Lauren L., Rebecca H., John C., Pat P.)
- [x] Create agenda
- [x] Round up content
- [x] Have meeting

**Comment by nickykrause** [comment]
Alright, so I pinged the FEC folks about what they are interested in knowing on Thurs. The big questions seem to be around the nuts and bolts of our process, rather than around the big-picture perspective of "What is UCD?" 

So, rather than a UCD 101 or Agile 101, they want to know what the cadence of our actual work looks like and how we collaborate between UX/Design to complete our design tasks.

💡  Here are the specific questions/ideas they provided:
> 1. I’m familiar with the design sprints process, but what tactics or methods do you employ to stay on schedule and on time with each sprint? The idea is to research, design, test, iterate – but outside of the theory, how do you actually achieve that in those restrictive timeframes?
> 2. For me, maybe just a general outline of the entire process so that when we get specific, I won’t lose sight of where that piece fits into the whole.
> 3.  I am interested in learning the mockup process using Invision and Sketch or whatever other tools we use. Of course, learning to use those programs is something more involved that I can do on my own with the advantage of some pro tips from 18F design team. But I think it will be helpful to know the workflow process that Jen and yourself go through to deliver mockups.

---

Questions 1 and 2 had me reflecting on our process and whether or not our actual work aligns with the "theory." I tried mapping out how our research/design/dev/testing cadence actually looks, and this is what I came up with: 

![artboard](https://cloud.githubusercontent.com/assets/1450173/24621071/d1dc890c-1865-11e7-9794-87d38fcb77fb.png)

I am trying to convey several things here:
- We have multiple feature streams happening simultaneously, and we try to off-set them to ensure that everyone has work to do from sprint to sprint
- We do iterate, but not necessarily all within one sprint. Sometimes we get through research/design in a sprint and then dev/testing in the next, followed by more testing/design, etc. It seems as if the "loops" that typically appear in diagrams of agile design/dev processes are actually drawn wider for us, across 2-3 sprints sometimes. 
- Some tasks are very large (like the red line) and we iterate over the design quite a bit before developing it, and, when the work moves to dev, it may be broken up into multiple pieces that are themselves iterated on as smaller features

I understand that this visual may make our process seem more linear or waterfall-esque than it really is. But, I am curious:
- ❓ @porta-antiporta and/or @jenniferthibault: What do you think about the idea I am trying to communicate? Does my understanding of the _reality_ of our process align with yours? (Note: I am not necessarily saying that this is the diagram we should use in the meeting; I am just trying to gather some input other than my own about how our process actually works in practice.)

--- 

In general, I think it would be helpful to use case studies as a means to answer their questions. Providing some kind of process diagram will probably be good, but a good example of how we have executed on the design process recently might be the most powerful means of answering 1, 2, and 3.  On that note
- @jenniferthibault: The case study that I am thinking might be useful is our recent redesign of the candidate & committee pages. Would you be on board with using that as a concrete example of our over-arching process, as well as the nuts and bolts of how we create mockups and collaborate?

**Comment by nickykrause** [comment]
Organizing some thoughts in [some slides here](https://docs.google.com/presentation/d/1Bka7Ip4vFdXFs1xK9VZDhBuagj_vwCNPfbvTu_uJZvs/edit#slide=id.p)

**Comment by porta-antiporta** [comment]
@nickykrause  i see your concern how the diagram above may seem agile-fall.  i have some ideas and comments which i'll place in the [slide deck].(https://docs.google.com/presentation/d/1Bka7Ip4vFdXFs1xK9VZDhBuagj_vwCNPfbvTu_uJZvs/edit#slide=id.p) 
cc: @jenniferthibault

**Comment by nickykrause** [comment]
Thanks, @porta-antiporta !

Quick update for all: I chatted with @jenniferthibault and @emileighoutlaw today, and we discussed a few things:
- Not all design tasks are created equal. 
    - Design tasks can be small, medium, large, or even extra-large.
    - We don't always follow all the same steps for every design task
    - We don't always complete steps in the same order (e.g., not everything starts with research, or even design -- sometimes tasks flow out of dev, etc.)
- Because not all tasks are created equal, our processes are flexible and reflexive, so there isn't really a one-size-fits-all diagram of _The Process_ (capital letters!). 
    - Instead of a set process, we follow certain principles (the "theory" behind what we do)
    - In reality, we adjust our approach to meet the needs of the situation. We make situation-dependent decisions throughout the design process that inform which steps we complete, how we complete them, and who is involved.
        - These decisions are informed by our principles (the "theory"), but, in practice, our work must also adapt to constraints (e.g., capacity, access to users, deadlines, etc.)

---
 
In light of all that, we thought it would be helpful to generate a list of the kinds of decisions we're making and how we make them (e.g., how do you know when to involve content in the design process?)
- I started a draft of those questions/answers [here](https://docs.google.com/document/d/1rFBbjSzETdWTndUwHOLuWcb744huYHzdOatQ7_wMFpM/edit#heading=h.3mn9kpcr1r31) :lock: 
    - Since the answer to these questions is often "it depends on the situation," we might show examples of several different kinds of situations and how we might address it, including examples.

We are also thinking that we can use case studies as a way of walking through how these questions/answers are visible in the designs we make -- we could show cases of S, M, L, and XL design tasks, and walk through what we did and why. 
- To get some ideas of the cases we might use, I added some of the ideas we discussed in [this document](https://docs.google.com/document/d/1Z4w32eRDNGUMeMfuT1woEz0cMq1qDfO25cpV73DT97g/edit) :lock:

Final thought: If we do show an "overall process diagram" of some kind, it should go at the end.

---

Next steps:
- I am going to continue fleshing out the docs mentioned above
- For whatever case studies we use, we could make InVision mockups to show how the design process looked for that case
- Jen and Emileigh and I are meeting again tomorrow to discuss

**Comment by nickykrause** [comment]
Meeting completed! 🎉 moving to done

**Comment by jenniferthibault** [comment]
@nickykrause @porta-antiporta , can this be closed?

**Comment by nickykrause** [comment]
I still need to upload the video for Lauren, but otherwise this meeting is finished, so I think we could close the issue.

It is worth asking, though, how we want to follow-up on some of the requested trainings that came up at the end of the session. Specifically, the group wanted to know:

- More about the nuts and bolts of working with our design tools and handing off between folks
- More info about how we manage our process (e.g., how GitHub fits into and structures our work)

We added them to the design pairing meetings in order to partially address the first one, but is there more we might do? 

As for the second one, I'm not sure. I am trying to recall specifically where the request came from, and I just remember Pat saying that GitHub was all she had really seen as a component of that process, and that it's "all she knows," but that there might be other methods that would work for their in-person team. 

Hashing out exactly how to address these follow-ups might not need to happen in this issue, but, before we close, I wanted to document them here. Shall we open up issues for those things? @porta-antiporta are these questions already covered any other issues related to the transition?

**Comment by jenniferthibault** [comment]
Agreed, the need for more granular trainings came up through this session—If at all possible, to keep focused I would be grateful if we could address those after the site flip & on-call period that follows the flip.

**Comment by porta-antiporta** [comment]
@jenniferthibault NO. it must happen BEFORE the site flip. PERIOD. _ducks_

What I suggest is we close this issue, have a discussion on what we think is the best way to engage the FEC folks into design activities (is it more granular trainings, or is it have them jump in the deep end of the pool and swim, combo of both).  From that discussion we can then target what stories we should be creating.

Looking at the mural from PI planning, we don't have explicit design transition stickies called out.  This is expected because I think we're getting to a level of detail (and information) that would not have been available during PI planning.

Would it be possible to have this discussion sometime next week? I think this would be a 30 min meeting w. @nickykrause @jenniferthibault @pat lauren and jonella.

**Comment by nickykrause** [comment]
@jenniferthibault and I just talked this morning with @patphongs. We had a sort of Q&A with Pat that was helpful for understanding what information would be useful for our next "walkthrough" session. 

We will look at scheduling a session for after the site flip, in which we can discuss some of our "design process management" practices, such as our standing meetings (grooming sessions, design pairings) and how we incorporate GitHub into our process.

I am closing this issue in favor of https://github.com/18F/fec-transition/issues/59, where we can plan that future session.

(cc @porta-antiporta)
